crates/services/operatorsinfo/src/operatorsinfo_inmemory.rs

```diff
@@ -146,20 +154,55 @@ impl OperatorInfoServiceInMemory {
                 while let Some(cmd) = pubkeys_rx.recv().await {
                     if let Err(e) = async {
                         match cmd {
-                            OperatorsInfoMessage::InsertOperatorInfo(addr, keys, socket_info) => {
+                            OperatorsInfoMessage::InsertOperatorInfo(
+                                addr,
+                                keys,
+                                socket_info,
+                                state_src,
+                            ) => {
                                 if let (Some(addr), Some(keys)) = (addr, keys) {
                                     let mut data = operator_state.operator_info_data.write().await;
-                                    data.insert(addr, *keys.clone());
+                                    match data.get(&addr) {
+                                        Some((entry_src, _))
+                                            if *entry_src == StateSource::Event
```

Comment:
can you add comment here that we don't overwrite latest data with historic one

Reply:
same at other places

Reply:
done